modular_skyrat/modules/customization/datums/dna.dm

```diff
@@ -180,7 +180,20 @@ GLOBAL_LIST_EMPTY(total_uf_len_by_block)
 	//We update the translation to make sure our character doesn't go out of the southern bounds of the tile
 	var/translate = ((change_multiplier-1) * 32)/2
 	holder.transform = holder.transform.Scale(change_multiplier)
-	holder.transform = holder.transform.Translate(0, translate)
+	// Check user's rotation before updating their translation, f.e. whether they're laying down or hanging upside down
+	var/translate_x = 0
+	var/translate_y = 0
+	if(holder.transform.a > 0)
+		translate_y = translate
+	else if(holder.transform.d > 0)
+		translate_x = -translate
+	else if(holder.transform.b > 0)
+		translate_x = translate
+	else if(holder.transform.a < 0)
+		translate_y = -translate
+	else
+		return
```

Comment:
What happens if I'm tilted by 45 degrees?

Reply:
I have not checked how the transform matrix changes when you turn anything other than 90, 180 or 270 degrees, so I genuinely don't know.

My guess is that one of the if-statements would still catch it. This **would** give unwanted results and probably break your sprite, but I don't think it'd be any worse than what you get without this PR.

A better way to solve it would be to create a math function based on transform.a, b, d and e for both translation_x and translation_y (so that they work for any rotation). I'll try to do that.

Reply:
Made functions based on transform.b and transform.e. Testing it right now and it appears to work fine with any angle, including 45 degrees.